compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -1313,6 +1321,59 @@ impl<'context> Elaborator<'context> {
         Ok(())
     }
 
+    /// Parses an attribute in the form of a function call (e.g. `#[foo(a b, c d)]`) into
+    /// the function and quoted arguments called (e.g. `("foo", vec![(a b, location), (c d, location)])`)
+    fn parse_attribute(
+        annotation: &str,
+        location: Location,
+    ) -> Option<(String, Vec<(Value, Location)>)> {
+        let (tokens, errors) = Lexer::lex(annotation);
+        if !errors.is_empty() {
```

Comment:
Can these errors be propagated? Currently, it appears to fail silently.

Reply:
They can't since we can't error for unknown attributes. Not all attributes are expected to be in the form of a function call for example. There aren't any I'd expect wouldn't be valid to lex but I think issuing these lexing errors would be more confusing than the "Unknown Attribute" error from the calling function here.